### ISSUE TITLE
PermutedDimsArray

### ISSUE THREAD

**Issue by antoine-levitt** [jipolanco/PencilArrays.jl#63]
Hi @jipolanco, just watching your talk now and have a technical question: for the custom permutated array that you seem to have, can't you just use PermutedDimsArray?

**Comment by antoine-levitt** [comment]
Ah sorry  I posted too fast, I thought you weren't going to come back to it but I see you did, sorry! Would be nice to have these improvements directly in julia Base!

**Comment by jipolanco** [comment]
I agree! And thanks for mentioning this. At some point I was planning to propose something like this for Julia base, but I wanted first to have a proper reference implementation of a `PermutedDimsArray` alternative.

The thing is that, for now, the `PencilArray` type includes code for dealing with permutations + all the other stuff that I presented. So the idea is to take the permutation stuff out of `PencilArray`, and define a `PermutedDimsArray`-like type which implements this. This type will likely be defined in the (very lightweight) [StaticPermutations.jl](https://github.com/jipolanco/StaticPermutations.jl) package, which already defines the `Permutation` type and a few permutation operations.
